src/psyclone/psyir/symbols/datasymbol.py

```diff
@@ -81,41 +81,23 @@ def resolve_deferred(self):
         ''' If the symbol has a deferred datatype, find where it is defined
         (i.e. an external container) and obtain the properties of the symbol.
 
-        :raises SymbolError: if the module pointed to by the symbol interface \
-                             does not contain the symbol (or the symbol is \
-                             not public).
-        :raises NotImplementedError: if the deferred symbol is not a Global.
+        :returns: this DataSymbol with its properties updated. This is for \
+                  consistency with the equivalent method in the Symbol \
+                  class which returns a new Symbol object.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
 
         '''
         from psyclone.psyir.symbols.datatypes import DeferredType
         if isinstance(self.datatype, DeferredType):
-            if self.is_global:
-                # Copy all the symbol properties but the interface and
-                # visibility (the latter is determined by the current
-                # scoping unit)
-                tmp = self.interface
-                module = self.interface.container_symbol
-                try:
-                    extern_symbol = module.container.symbol_table.lookup(
-                        self.name, visibility=Symbol.Visibility.PUBLIC)
-                except KeyError:
-                    raise SymbolError(
```

Comment:
SymbolError is now unused and the import needs to be deleted

Reply:
Oh yes, thanks.